ethcore/src/engines/authority_round/mod.rs

```diff
@@ -597,12 +755,32 @@ impl Engine<EthereumMachine> for AuthorityRound {
 				return Seal::None;
 			}
 
+			// FIXME: configurable
+			const MAX_EMPTY_STEPS: usize = 15;
+
+			// if there are no transactions to include in the block, we don't seal and instead broadcast a signed
+			// `EmptyStep(step)` message. If we exceed the maximum amount of `empty_step` rounds we proceed with the
+			// seal.
+			if block.transactions().is_empty() && empty_steps.len() < MAX_EMPTY_STEPS {
+				self.generate_empty_step(header.parent_hash());
+				return Seal::None;
+			}
+
 			if let Ok(signature) = self.sign(header.bare_hash()) {
 				trace!(target: "engine", "generate_seal: Issuing a block for step {}.", step);
 
 				// only issue the seal if we were the first to reach the compare_and_swap.
 				if self.can_propose.compare_and_swap(true, false, AtomicOrdering::SeqCst) {
-					return Seal::Regular(vec![encode(&step).into_vec(), encode(&(&H520::from(signature) as &[u8])).into_vec()]);
+
+					// clear old `empty_steps` messages
+					self.empty_steps.lock().retain(|e| U256::from(e.step) > parent_step);
+					let fields = vec![
+						encode(&step).into_vec(),
+						encode(&(&H520::from(signature) as &[u8])).into_vec(),
+						::rlp::encode_list(&empty_steps).into_vec(),
```

Comment:
I'm having some trouble with this, for some reason when the other nodes sync the block they can't deserialize the seal, it actually fails validation as they see a larger number of fields in the seal than expected (3).

Reply:
check the `Engine::seal_fields` function is returning the right count

Reply:
I updated the `seal_fields` function to return the correct number of fields. When deserializing the other nodes see a variable number of fields which I think varies with the size of the `empty_steps` list.

> 2018-01-15 17:52:23 UTC IO Worker #3 WARN client  Stage 1 block verification failed for 50e0…c519: Block(InvalidSealArity(Mismatch { expected: 3, found: 11 }))